packages/zoe/src/offerSafety.js

```diff
@@ -1,38 +1,93 @@
 /**
- * `isOfferSafeForOffer` checks offer safety for a single offer.
- *
- * Note: This implementation checks whether we refund for all rules or
- * return winnings for all rules. It does not allow some refunds and
- * some winnings, which is what would happen if you checked the rules
- * independently. It *does* allow for returning a full refund plus
- * full winnings.
- *
- * @param  {function} getAmountMath - a function that takes a brand and returns
- * the appropriate amountMath
- * @param  {object} proposal - the rules that accompanied the
- * escrow of payments that dictate what the user expected to get back
- * from Zoe. A proposal is a record with keys `give`,
- * `want`, and `exit`. `give` and `want` are records with keywords
- * as keys and amounts as values. The proposal is a player's
- * understanding of the contract that they are entering when they make
- * an offer.
- * @param  {object} newAmountKeywordRecord - a record with keywords as keys and
- * amounts as values. These amounts are the reallocation to be given to a user.
+ * @typedef {import('@agoric/ertp/src/issuer').Brand} Brand
+ * @typedef {import('@agoric/ertp/src/issuer').AmountMath} AmountMath
+ * @typedef {import('./zoe').Proposal} Proposal
+ * @typedef {import('./zoe').AmountKeywordRecord} AmountKeywordRecord
+ */
+
+/**
+ * Helper to perform satisfiesWant and satisfiesGive. Is
+ * allocationAmount greater than or equal to requiredAmount for every
+ * keyword of giveOrWant?
+ * @param {(Brand) => AmountMath} getAmountMath
+ * @param {Proposal["give"] | Proposal["want"]} giveOrWant
+ * @param {AmountKeywordRecord} allocation
+ */
+const satisfies = (getAmountMath, giveOrWant, allocation) => {
```

Comment:
Could this be `satisfiesInternal` or something else distinctive. I was a little confused by its relationship with satisfies in zoeHelpers.js.

Reply:
Done